### PR TITLE
CMakeLists.txt: Un-prefix absolute install dirs

### PR DESCRIPTION
**Pull request by tari01**
fixes https://github.com/AyatanaIndicators/ayatana-ido/issues/40

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,27 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
...
+endfunction()
+
+unprefix(CMAKE_INSTALL_BINDIR)
+unprefix(CMAKE_INSTALL_LIBEXECDIR)
+unprefix(CMAKE_INSTALL_LIBDIR)
+unprefix(CMAKE_INSTALL_INCLUDEDIR)
+unprefix(CMAKE_INSTALL_DATADIR)
+unprefix(CMAKE_INSTALL_LOCALEDIR)
+unprefix(CMAKE_INSTALL_MANDIR)
+unprefix(CMAKE_INSTALL_DOCDIR)
```
Comment (sunweaver) [inline CMakeLists.txt:29]: Why undo stuff that is actually seemingly (haven't tested, just read on the web) fine?

Simply don't hard-coded-ly prepent CMAKE_INSTALL_PREFIX and then we should be done.

**Review by sunweaver — CHANGES_REQUESTED** [review]
This looks like a hack-around around something that is well covered by CMake. Isn't it?

File: CMakeLists.txt (hunk 1 above)
Comment (tari01) [reply CMakeLists.txt:29]: CMake does **NOT** prepend `CMAKE_INSTALL_PREFIX` to `CMAKE_INSTALL_*DIR` variables (tested it). The expected behaviour is to provide an absolute path to `CMAKE_INSTALL_PREFIX`, and relative to that, all the `CMAKE_INSTALL_*DIR`. It is not an error to use absolute paths, but it should be then handled somehow. 

This is not so much a hack, but a way to accommodate that 1% of distros that do this. Normally, we would depend on `CMAKE_INSTALL_FULL__*DIR`, but what if they override that as well?

http://cmake.3232098.n2.nabble.com/CMAKE-INSTALL-LIBDIR-full-path-or-relative-td7302556.html
https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

There are two solutions:
1. Use this sanity-fix in every project
2. Use `CMAKE_INSTALL_FULL__*DIR` in install funtions and hope for the best

Circle the desired option.

Comment (sunweaver) [reply CMakeLists.txt:29]: What I normally see, and I mixed it up, is CMAKE_INSTALL_FULL_* in install target rules. So circle is made around No. 2.

Comment (tari01) [reply CMakeLists.txt]: > What I normally see, and I mixed it up, is CMAKE_INSTALL_FULL_* in install target rules. So circle is made around No. 2.

Done.

**Review by sunweaver — APPROVED** [review]
LGTM

**Pull request merged**